src/v/ssx/tests/single_sharded_test.cc

```diff
@@ -80,7 +80,7 @@ struct caller {
 
 SEASTAR_THREAD_TEST_CASE(single_sharded) {
     ss::shard_id the_shard = ss::smp::count - 1;
-    ss::shard_id wrong_shard = std::max(ss::shard_id{0}, the_shard - 1);
+    ss::shard_id wrong_shard = ss::smp::count == 1 ? 0 : the_shard - 1;
```

Comment:
but if we run with `-c1`, there is no wrong shard, right? So I guess it would be more obvious if the part of the test requiring several shards would be clearly encapsulated in a code block (or maybe even a separate test) and `wrong_shard` would be declared there

Reply:
Can you think of an existing test that _requires_ multiple shards? What does it do if there is only one shard?

Reply:
I think its ok to assume a minimum shard count in tests, just have an assert in the test, something like

```
BOOST_REQUIRE_GT(ss::smp::count, 1)
```

There are tests in the tree that does this, better to assert out than a segfault.